src/test/java/org/folio/rest/impl/EholdingsProvidersImplTest.java

```diff
@@ -0,0 +1,139 @@
+import static org.hamcrest.Matchers.notNullValue;
+
+@RunWith(VertxUnitRunner.class)
+public class EholdingsProvidersImplTest {
+  private static final Header TENANT_HEADER = new Header(RestConstants.OKAPI_TENANT_HEADER, "fs");
+  private static final Header TOKEN_HEADER = new Header(RestConstants.OKAPI_TOKEN_HEADER, "TEST_OKAPI_TOKEN");
+  private static final String STUB_CUSTOMER_ID = "TEST_CUSTOMER_ID";
+  private static RequestSpecification spec;
+  private static int port;
+  private static String host;
+
+  @org.junit.Rule
+  public WireMockRule userMockServer = new WireMockRule(
+    WireMockConfiguration.wireMockConfig()
+      .dynamicPort()
+      .notifier(new ConsoleNotifier(true)));
+
+  @BeforeClass
+  public static void setUpClass(final TestContext context) {
+    Vertx vertx = Vertx.vertx();
+    vertx.exceptionHandler(context.exceptionHandler());
+    port = NetworkUtils.nextFreePort();
+    host = "http://localhost";
+
+    DeploymentOptions restVerticleDeploymentOptions = new DeploymentOptions()
+      .setConfig(new JsonObject().put("http.port", port));
+    vertx.deployVerticle(RestVerticle.class.getName(), restVerticleDeploymentOptions, context.asyncAssertSuccess());
+
+    spec = new RequestSpecBuilder()
+      .setBaseUri(host + ":" + port)
+      .build();
+  }
+
+  @Before
+  public void setUp() {
+    RMAPIConfigurationCache.getInstance().invalidate();
+  }
+
+  @Test
+  public void shouldReturnProvidersOnGet() throws IOException, URISyntaxException {
+    String stubResponseFile = "responses/rmapi/vendors/get-vendors-response.json";
+    int expectedTotalResults = 115;
+    String id = "131872";
+    String name = "Editions de L'Université de Bruxelles";
+    int packagesTotal = 1;
+    int packagesSelected = 0;
+    boolean supportsCustomPackages = false;
+
+    String wiremockUrl = host + ":" + userMockServer.port();
+    TestUtil.mockConfiguration("responses/configuration/get-configuration.json", wiremockUrl);
+    WireMock.stubFor(
+      WireMock.get(new UrlPathPattern(new RegexPattern("/rm/rmaccounts/" + STUB_CUSTOMER_ID + "/vendors.*"), true))
+        .willReturn(new ResponseDefinitionBuilder()
+          .withBody(TestUtil.readFile(stubResponseFile))));
+    RestAssured.given()
+      .spec(spec).port(port)
+      .header(new Header(RestConstants.OKAPI_URL_HEADER, wiremockUrl))
+      .header(TENANT_HEADER)
+      .header(TOKEN_HEADER)
+      .when()
+      .get("eholdings/providers?q=e&page=1&sort=name")
+      .then()
+      .statusCode(200)
+      .body("meta.totalResults" , equalTo(expectedTotalResults))
+      .body("data[0].type", equalTo("providers"))
+      .body("data[0].id", equalTo(id))
+      .body("data[0].attributes.name", equalTo(name))
+      .body("data[0].attributes.packagesTotal", equalTo(packagesTotal))
+      .body("data[0].attributes.packagesSelected", equalTo(packagesSelected))
+      .body("data[0].attributes.supportsCustomPackages", equalTo(supportsCustomPackages));
```

Comment:
Also, just curious, why are we not checking for "providerToken" in response even if it ends up being null?